ssl/quic/quic_demux.c

```diff
@@ -460,20 +460,22 @@ static int demux_recv(QUIC_DEMUX *demux)
 /* Extract destination connection ID from the first packet in a datagram. */
 static int demux_identify_conn_id(QUIC_DEMUX *demux,
                                   QUIC_URXE *e,
-                                  QUIC_CONN_ID *dst_conn_id)
+                                  QUIC_CONN_ID *dst_conn_id,
+                                  int *pkt_too_short)
 {
     return ossl_quic_wire_get_pkt_hdr_dst_conn_id(ossl_quic_urxe_data(e),
                                                   e->data_len,
                                                   demux->short_conn_id_len,
-                                                  dst_conn_id);
+                                                  dst_conn_id, pkt_too_short);
 }
 
 /* Identify the connection structure corresponding to a given URXE. */
-static QUIC_DEMUX_CONN *demux_identify_conn(QUIC_DEMUX *demux, QUIC_URXE *e)
+static QUIC_DEMUX_CONN *demux_identify_conn(QUIC_DEMUX *demux, QUIC_URXE *e,
+                                            int *pkt_too_short)
 {
     QUIC_CONN_ID dst_conn_id;
 
-    if (!demux_identify_conn_id(demux, e, &dst_conn_id))
+    if (!demux_identify_conn_id(demux, e, &dst_conn_id, pkt_too_short))
```

Comment:
If a packet is sufficiently malformed that `ossl_quic_wire_get_pkt_hdr_dst_conn_id` fails, we may as well reject processing of it in any circumstance. So I would suggest getting rid of the changes in `quic_wire_pkt.c` and just modifying this function in DEMUX to indicate whether it's a failure of `demux_identify_conn_id` or a lookup failure. Flag can be a more general `pkt_invalid` or similar.

Reply:
I think this is required.   As per RFC 9000 s. 10.3
> Endpoints MUST discard packets that are too small to be valid QUIC packets. To give an example, with the set of AEAD functions defined in [[QUIC-TLS](https://datatracker.ietf.org/doc/html/rfc9001)], short header packets that are smaller than 21 bytes are never valid.

Without knowing that the packet is undersized, the code in demux_process_pending_urxe() will pass it to the default packet handler for processing.  This must not occur and the packet must be forcibly dropped.

https://github.com/openssl/openssl/blob/06a0d40322e96dbba816b35f82226871f635ec5a/ssl/quic/quic_demux.c#L499-L514

I do like the idea of making this a flags argument instead.

Reply:
The QRX rejects any packet which is too small to be a valid encrypted packet.

Any handling in the default packet handler will begin with some specific unmarshalling which will enforce any necessary packet geometry as an implication of its operation. So we do not need to do anything in the DEMUX to meet this requirement.